### ISSUE TITLE
Feature: Listen / Notify support

### ISSUE THREAD

**Issue by TomMD** [valderman/selda#128]
It appears SQLite has sufficient functionality for a listen/notify framework (https://www.sqlite.org/c3ref/update_hook.html).  Postgresql certainly does (ex. https://www.postgresql.org/docs/9.0/sql-notify.html).

What is your thinking about supporting a listen/notify functionality in Selda?

**Comment by valderman** [comment]
Supporting either of those APIs wouldn't be hard, I think. However, since they do relatively different things, it's not entirely clear what sort of functionality we'd actually want from them, and what the API should look like.

Do you have a particular use case in mind, which could give some additional insight into what this support might look like?

**Comment by TomMD** [comment]
I do have a particular case in mind.  For me specifically, I'm designing towards a managed postgresql where a component inserts values into table A (via one transaction) which triggers (via listen/notify) another component to perform a computational pass which reads tables A and B to populate table C.

I recognize that any selda support can not leave sqlite behind, but the update hook of sqlite can simulate the listen/notify behavior (I believe).

**Comment by valderman** [comment]
Since we now finally have the ability to support per-backend functionality in place, I don't think it'd be a big deal to initially support this for PostgreSQL with an appropriate type class constraint and work out the details for SQLite later. In particular, `direct-sqlite` doesn't currently support update hooks, so we'd have to get that support merged first.

For the API for a feature like this, I'm imagining something like:
```
newChannel :: (MonadSelda m, NotifyBackend b) => m b (NotifyChannel b)
notify :: (MonadSelda m, NotifyBackend b) => Maybe Text -> NotifyChannel b -> m b ()
listen :: (MonadSelda m, NotifyBackend b) => NotifyChannel b -> m b (Maybe Text)
```
...where `listen` would block until a notification arrives. Thoughts?

One issue here would be what happens when the user tries to `listen` on the wrong database. As we don't encode the database in the backend type parameter (and for the record I don't think we should, at least not by default, or it'd be extremely cumbersome to do things like dynamic configuration), we'd basically have to introduce a new `SeldaError` for this. It's workable but not very elegant, and a safer solution to this problem would be nice to have.

**Comment by TomMD** [comment]
Yes, using a class for Notify could work well.  It provides a way forward that doesn't break the other backends or restrain selda to the lowest common denominator backend.

I'll consider the issue of listening on a non-existent DB (that is the concern, right?) but suspect the listen could just return an either unless I missed something.
